vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt

```diff
@@ -208,6 +179,34 @@ class LocationSharingService : VectorService(), LocationTracker.Callback {
         destroyMe()
     }
 
+    @Synchronized
+    private fun addRoomArgs(beaconEventId: String, roomArgs: RoomArgs) {
+        roomArgsMap[beaconEventId] = roomArgs
+    }
+
+    @Synchronized
+    private fun removeRoomArgs(roomId: String) {
+        val beaconIds = roomArgsMap
+                .filter { it.value.roomId == roomId }
+                .map { it.key }
+        beaconIds.forEach { roomArgsMap.remove(it) }
+    }
+
+    private fun listenForLiveSummaryChanges(roomId: String, eventId: String) {
+        activeSessionHolder
+                .getSafeActiveSession()
+                ?.let { session ->
+                    mainHandler.post {
```

Comment:
Do we need `mainHandler` here? We generally use it while triggering a listener function.

Reply:
See [this comment](https://github.com/vector-im/element-android/pull/6356#discussion_r906047460) for explanation. Since we are in Android Service, this is the only way I found to access main thread.

Reply:
this is where I'm a little confused, the android service should already be calling back on the main thread, we're jumping to the IO thread here https://github.com/vector-im/element-android/blob/81e14c7c3b6529f08d2cf0b0cacc5a548c836c12/vector/src/main/java/im/vector/app/features/location/LocationSharingService.kt#L89 :thinking:

Reply:
Yes you are right. Maybe we whould not switch the `Dispatcher` and rely on the fact the SDK is handling correctly the Network calls. I will try to remove it to see.

Reply:
It seems to work. I will remove the `Handler` I have introduced and the switch to the `Dispatcher.io`.